app/qml/map/RecordCrosshair.qml

```diff
@@ -8,23 +8,233 @@
+      y: root.recordPoint.y - height / 2
+
+      Behavior on x {
+        PropertyAnimation {
+          properties: "x"
+          duration: 50
+          easing.type: Easing.InQuad
+        }
+      }
+
+      Behavior on y {
+        PropertyAnimation {
+          properties: "y"
+          duration: 50
+          easing.type: Easing.InQuad
+        }
+      }
+
+      opacity: snapUtils.snapped ? 0 : 100
+
+      Behavior on opacity {
+        PropertyAnimation {
+          properties: "opacity"
+          duration: 100
+          easing.type: Easing.InQuad
+        }
+      }
+
+      height: root.innerDotSize
+      width: height
+      sourceSize.width: width
+      sourceSize.height: height
+
+      source: InputStyle.crosshairCenterDotIcon
+    }
+
+    Image {
+      id: crossCenterPlus // Center dot - visible when not snapped
+
+      x: root.recordPoint.x - width / 2
+      y: root.recordPoint.y - height / 2
+
+      Behavior on x {
+        PropertyAnimation {
+          properties: "x"
+          duration: 50
+          easing.type: Easing.InQuad
+        }
+      }
+
+      Behavior on y {
+        PropertyAnimation {
+          properties: "y"
+          duration: 50
+          easing.type: Easing.InQuad
+        }
+      }
+
+      opacity: snapUtils.snapped && ( snapUtils.snapType === SnapUtils.Vertex || snapUtils.snapType === SnapUtils.Other ) ? 100 : 0
+
+      Behavior on opacity {
+        PropertyAnimation {
+          properties: "opacity"
+          duration: 100
+          easing.type: Easing.InQuad
+        }
+      }
+
+      rotation: snapUtils.snapType === SnapUtils.Other ? 0 : 45
+
+      Behavior on rotation {
+        PropertyAnimation {
+          properties: "rotation"
+          duration: 50
+          easing.type: Easing.InQuad
+        }
+      }
+
+      height: root.innerDotSize * 2
+      width: height
+      sourceSize.width: width
+      sourceSize.height: height
+
+      source: InputStyle.crosshairCenterPlusIcon
+    }
+
+    Image {
+      id: crossCenterCircle // Center circle - visible when snapped to segment
+
+      x: root.recordPoint.x - width / 2
+      y: root.recordPoint.y - height / 2
+
+      Behavior on x {
+        PropertyAnimation {
+          properties: "x"
+          duration: 50
+          easing.type: Easing.InQuad
+        }
+      }
+
+      Behavior on y {
+        PropertyAnimation {
+          properties: "y"
+          duration: 50
+          easing.type: Easing.InQuad
+        }
+      }
+
+      opacity: snapUtils.snapped && snapUtils.snapType === SnapUtils.Segment ? 100 : 0
+
+      Behavior on opacity {
+        PropertyAnimation {
+          properties: "opacity"
+          duration: 100
+          easing.type: Easing.InQuad
+        }
+      }
+
+      height: root.innerDotSize * 2
+      width: height
+      sourceSize.width: width
+      sourceSize.height: height
+
+      source: InputStyle.crosshairCenterCircleIcon
+    }
+
+    Connections {
```

Comment:
This connection to `_digitizingController` works just because the `RecordCrosshair` is instantiated within `MapWrapper`, however, if we would move the `RecordCrosshair` somewhere else, this is logic is going to break.

I suggest to create a new property in `RecordCrosshair` something like `shouldUseSnapping` and inside `MapWrapper` bind it to:

```qml
RecordCrosshair {
  id: recordCrosshair
  ...
  shouldUseSnapping: _digitizingController.manualRecording && !_digitizingController.useGpsPoint
}
```
instead.

That way if we would move the `RecordCrosshair` component somewhere else we would just need to fix the binding 🙂